routes/challenge.js

```diff
@@ -186,6 +189,51 @@ exports.edit = function(req, res) {
 };
 
 /*
+Change challenge formulae by which users are quantified.
+*/
+exports.update_formulae = function(req, res) {
+
+  Challenges.findOne({'link': req.params.ch}).exec(gotChallenge)
+
+  function gotChallenge(err, ch) {
+    // Check if user is admin
+    if (ch.admins.indexOf(req.session.auth.github.user.login) < 0)
+      return res.redirect('/challenges/' + req.params.ch)
+
+
+    display_scores = false
+    if (req.body.display_scores == 'on')
+      display_scores = true
+
+    var conditions = {'link': req.params.ch};
+    var update = {$set: {
+      'formulae': req.body.formulae,
+      'display_scores': display_scores
+    }}
+    Challenges.update(conditions, update, function (err, num) {
```

Comment:
I don't understand what this block of code does. Why do you define the above conditions / update, since you actually rewrite and apply them inside the method?